librz/include/rz_types.h

```diff
@@ -333,6 +333,7 @@ static inline void *rz_new_copy(int size, void *data) {
 }
 // TODO: Make RZ_NEW_COPY be 1 arg, not two
 #define RZ_NEW_COPY(x,y) x=(void*)malloc(sizeof(y));memcpy(x,y,sizeof(y))
+#define RZ_MEMCPY_S(dest, dmax, src, smax) memcpy (dest, src, (smax < dmax) ? smax : dmax)
```

Comment:
Could you please add here an `#ifdef` with MSVC and newer GCC/Glibc (C11 should have support for `memcpy_s()`):
- https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/memcpy-s-wmemcpy-s?view=msvc-160
- https://en.cppreference.com/w/c/string/byte/memcpy

Reply:
Moreover, we can make an inline function and put it in the utils instead. It will allow to check the types of the arguments, that `smax`, `dmax` should be size_t, not a signed integer, etc.

Reply:
Actually, instead of relying on checking various macros... Just use `meson` to check if the function is available. You can look at https://github.com/rizinorg/rizin/blob/dev/meson.build#L355 . That is imho the best way to see if some functions/features are available in the compiler.

Reply:
I stayed with regular memcpy since memcpy_s has a different return value and we are going to set 99 anyways(?), I don't see a reason to split the implementation with ifdefs when `rz_mem_copy` does the same thing.